auth-service/src/test/kotlin/uk/gov/communities/delta/controllers/EditAccessGroupsControllerTest.kt

```diff
@@ -0,0 +1,426 @@
+            contentType(ContentType.Application.Json)
+            setBody(
+                "{" +
+                        "\"accessGroupsRequest\": {\"datamart-delta-access-group-1\": [\"orgCode1\", \"orgCode2\"], \"datamart-delta-access-group-2\": [], \"datamart-delta-access-group-3\": []}" +
+                        ", \"userSelectedOrgs\": [\"orgCode1\", \"orgCode2\"]" +
+                        "}"
+            )
+        }.apply {
+            assertEquals(HttpStatusCode.OK, status)
+            coVerify(exactly = 1) {
+                groupService.addUserToGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-1-orgCode2",
+                    any(),
+                    null
+                )
+            }
+            coVerify(exactly = 1) {
+                groupService.removeUserFromGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-2",
+                    any(),
+                    null
+                )
+            }
+            coVerify(exactly = 1) {
+                groupService.removeUserFromGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-2-orgCode1",
+                    any(),
+                    null
+                )
+            }
+            coVerify(exactly = 1) {
+                groupService.removeUserFromGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-3",
+                    any(),
+                    null
+                )
+            }
+            coVerify(exactly = 1) {
+                groupService.removeUserFromGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-3-orgCode2",
+                    any(),
+                    null
+                )
+            }
+            confirmVerified(groupService)
+        }
+    }
+
+    @Test
+    fun accessGroupIsAddedIfUserIsntAlreadyMember() = testSuspend {
+        val accessGroupsRequestMap = mapOf("ag1" to listOf("org1"))
+        val currentAccessGroupsMap = mapOf<String, List<String>>()
+        val selectedOrgs = setOf("org1")
+        val actions =
+            controller.generateAccessGroupActionList(accessGroupsRequestMap, currentAccessGroupsMap, selectedOrgs)
+        assertTrue {
+            actions.filter {
+                it.getActiveDirectoryString()
+                    .equals("datamart-delta-ag1-org1") && it is AddAccessGroupOrganisationAction
+            }.size == 1
+        }
+        assertTrue {
+            actions.filter {
+                it.getActiveDirectoryString()
+                    .equals("datamart-delta-ag1") && it is AddAccessGroupAction
+            }.size == 1
+        }
+        assertTrue { actions.size == 2 }
```

Comment:
If we make actions a set then

```suggestion
        assertEquals(
            setOf(
                AddAccessGroupAction("ag1"),
                AddAccessGroupOrganisationAction("ag1", "org1")
            ),
            actions
        )
```